libecs/source/Core/Scene.cxx

```diff
@@ -74,3 +74,21 @@ const pivot::ecs::systems::Manager &Scene::getSystemManager() const { return mSy
 pivot::ecs::event::Manager &Scene::getEventManager() { return mEventManager; }
 
 const pivot::ecs::event::Manager &Scene::getEventManager() const { return mEventManager; }
+
+void Scene::save(const std::filesystem::path &path)
+{
+    // serialize scene
+    nlohmann::json output;
+    output["name"] = name;
+    for (auto &[entity, _]: mEntityManager.getEntities())
+        for (pivot::ecs::component::ComponentRef ref: mComponentManager.GetAllComponents(entity))
+            output["components"][entity][ref.description().name] = nlohmann::json(ref.get());
+    std::vector<std::string> systems;
+    for (auto &[systemName, _]: mSystemManager) { systems.push_back(systemName); }
+    output["systems"] = systems;
+    // write in file
+    if (!std::filesystem::exists(path)) { std::filesystem::create_directories(path.parent_path()); }
```

Comment:
No need to create the directory here. This should be handled by the engine or the file selection dialog.
```suggestion
```